### PR TITLE
Check that the current version of CMake supports policy 141 before se…

### LINKED ISSUE #1919: [BUG]: Policy CMP0141 is not known to CMake<3.25
### Is this a duplicate?

- [X] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Something else

### Component

Not sure

### Describe the bug

Using CCCL with MSVC and CMake<3.25 produces the following error during CMake configuration:
```
CMake Error at cccl/CMakeLists.txt:7 (cmake_policy):
  Policy "CMP0141" is not known to this version of CMake.
```
My CMake version is 3.22, so I believe this also affects the developer build.

### How to Reproduce

1. clone the CCCL repo into some CMake project
2. add CCCL to the CMake project using `add_subdirectory`
3. try to configure the project

### Expected behavior

The configuration step of the CMake project should work without producing this error.

### Reproduction link

_No response_

### Operating System

Windows 10 Pro

### nvidia-smi output

_No response_

### NVCC version

_No response_

### PR DESCRIPTION
**Pull request by alliepiper**
Fixes #1919

### CODE REVIEW

**Review by miscco — APPROVED** [review]
I am ok with this change, if users use old CMAKE they can take the hit of sccache not working properly

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@
 cmake_minimum_required(VERSION 3.15)
 
 # sccache cannot handle the -Fd option generationg pdb files
-if ("MSVC" STREQUAL "${CMAKE_CXX_COMPILER_ID}")
+if ("MSVC" STREQUAL "${CMAKE_CXX_COMPILER_ID}" AND CMAKE_VERSION VERSION_GREATER_EQUAL 3.25)
   cmake_policy(SET CMP0141 NEW)
 endif()
```
Comment (bernhardmgruber) [inline CMakeLists.txt:8]: Question: Why cannot we write:
```suggestion
if ("MSVC" STREQUAL "${CMAKE_CXX_COMPILER_ID}" AND POLICY CMP0141)
  cmake_policy(SET CMP0141 NEW)
endif()
```
CMake has a conditional to check if a policy exists: https://cmake.org/cmake/help/latest/command/if.html#policy

Comment (alliepiper) [reply CMakeLists.txt:8]: Nice catch! I didn't know about this one.

Comment (alliepiper) [reply CMakeLists.txt:8]: Also dropping the compiler check since this policy only affects MSVC.

Comment (bernhardmgruber) [reply CMakeLists.txt]: Great!

**Review by bernhardmgruber — APPROVED** [review]

**Pull request merged**